packages/forms/src/index.ts

```diff
@@ -212,6 +212,87 @@ export const addPatternToPage = (
   };
 };
 
+export const movePatternBetweenPages = (
+  bp: Blueprint,
+  sourcePageId: PatternId,
+  targetPageId: PatternId,
+  patternId: PatternId,
+  position: string,
+  isPageMove: boolean = false
+): Blueprint => {
+  const sourcePage = bp.patterns[sourcePageId] as PagePattern;
+  const targetPage = bp.patterns[targetPageId] as PagePattern;
+
+  if (!sourcePage || !targetPage) {
+    throw new Error('Source or target page not found.');
+  }
+
+  if (sourcePage.type !== 'page' || targetPage.type !== 'page') {
+    throw new Error('Pattern is not a page.');
+  }
+
+  let updatedSourcePatterns: PatternId[];
+  let updatedTargetPatterns: PatternId[];
+
+  if (isPageMove) {
+    const sourcePagePatterns = sourcePage.data.patterns;
+    const indexToRemove = sourcePagePatterns.indexOf(patternId);
+
+    if (indexToRemove === -1) {
+      throw new Error(`Pattern ID ${patternId} not found in the source page.`);
+    }
+
+    const newPattern = sourcePagePatterns.splice(indexToRemove, 1).toString();
+
+    updatedSourcePatterns = [
+      ...sourcePagePatterns.slice(0, indexToRemove),
+      ...sourcePagePatterns.slice(indexToRemove + 1),
+    ];
+
+    updatedTargetPatterns =
+      position === 'top'
+        ? [newPattern, ...updatedSourcePatterns]
```

Comment:
I could be confused, but I think you can do this, instead of reading the value from the return value of the splice.

```suggestion
        ? [patternId, ...updatedSourcePatterns]
```

Reply:
I ended up not using newPattern anymore and removed it.